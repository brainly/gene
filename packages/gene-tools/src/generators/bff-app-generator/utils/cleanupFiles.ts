import { Tree } from '@nx/devkit';
import { BrainlyNextJSAppGenerator } from '../schema';

export const cleanupFiles = (tree: Tree, filesToRemove: Array<string>) => {
  filesToRemove &&
    tree
      .listChanges()
      .map(({ path }) => path)
      .filter((path) => filesToRemove.some((e) => path.endsWith(e)))
      .forEach((path) => tree.delete(path));
};
