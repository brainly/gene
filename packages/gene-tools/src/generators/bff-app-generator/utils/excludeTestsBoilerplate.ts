import {Tree} from '@nrwl/devkit';

export const excludeTestsBoilerplate = (tree: Tree) => {
  const e2eBoilerplatePaths = tree
    .listChanges()
    .map(({path}) => path)
    .filter(
      path =>
        path.endsWith('support/app.po.ts') ||
        path.endsWith('integration/app.spec.ts') ||
        path.endsWith('index.spec.tsx')
    );

  e2eBoilerplatePaths.forEach(path => tree.delete(path));
};
