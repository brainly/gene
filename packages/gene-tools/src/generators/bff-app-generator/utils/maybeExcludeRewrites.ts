import type { Tree } from '@nx/devkit';
import type { BrainlyNextJSAppGenerator } from '../schema';

export const maybeExcludeRewrites = (
  tree: Tree,
  schema: BrainlyNextJSAppGenerator
) => {
  if (!schema.rewrites) {
    const rewriteFilePath = tree
      .listChanges()
      .map(({ path }) => path)
      .find(
        (path) =>
          path.endsWith('rewrites.json') || path.endsWith('loadRewrites.js')
      );

    if (rewriteFilePath) {
      tree.delete(rewriteFilePath);
    }
  }
};
