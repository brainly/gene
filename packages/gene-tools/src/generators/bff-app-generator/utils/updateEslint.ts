import { joinPathFragments, Tree, updateJson } from '@nx/devkit';

export const updateEslint = (tree: Tree, appDir: string) => {
  updateJson(
    tree,
    joinPathFragments(appDir, './.eslintrc.json'),
    (eslintConfig) => {
      return {
        ...eslintConfig,
        settings: {
          next: {
            rootDir: appDir,
          },
        },
      };
    }
  );
};
