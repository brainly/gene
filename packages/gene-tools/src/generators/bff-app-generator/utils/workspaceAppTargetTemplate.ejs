{
  "build": {
    "dependsOn": [
     {
        "target": "copy-translations-to-app",
        "projects": "self"
     },
     {
        "target": "copy-assets-to-app",
        "projects": "self"
      },
      {
        "target": "build",
        "projects": "dependencies"
      }
    ],
    "executor": "@nx/next:build",
    "outputs": ["{options.outputPath}"],
    "options": {
      "root": "apps/<%= projectPath %>",
      "outputPath": "dist/apps/<%= projectPath %>"
    },
    "configurations": {
      "production": {}
    }
  },
  "serve-base": {
    "dependsOn": [
       {
          "target": "copy-translations-to-app",
          "projects": "self"
       },
       {
          "target": "copy-assets-to-app",
          "projects": "self"
        }
    ],
    "executor": "@nx/next:server",
    "options": {
      "buildTarget": "<%= projectName %>:build",
      "dev": true
    },
    "configurations": {
      "production": {
        "buildTarget": "<%= projectName %>:build:production",
        "dev": false
      }
    }
  },
  "serve": {
    "executor": "@brainly-gene/tools:secure-serve",
    "options": {
      "serveTarget": "<%= projectName %>:serve-base"
    },
    "dependsOn": [
     {
        "target": "copy-translations-to-app",
        "projects": "self"
     },
     {
        "target": "copy-assets-to-app",
        "projects": "self"
      },
      {
        "target": "build",
        "projects": "dependencies"
      }
    ],
    "configurations": {
      "production": {
        "serveTarget": "<%= projectName %>:serve-base:production"
      }
    }
  },
  "copy-translations-to-app": {
    "executor": "@brainly-gene/tools:copy-translations-to-app"
  },
  "copy-assets-to-app": {
    "executor": "@brainly-gene/tools:copy-assets-to-app"
  }
}
