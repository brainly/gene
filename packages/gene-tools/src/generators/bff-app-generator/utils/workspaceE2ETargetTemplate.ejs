{
  "e2e": {
    "executor": "@nx/cypress:cypress",
    "options": {
      "cypressConfig": "apps/<%= projectPath %>-e2e/cypress.config.ts",
      "testingType": "e2e",
      "devServerTarget": "<%= projectName %>:serve-base"
    },
    "configurations": {
      "production": {
        "devServerTarget": "<%= projectName %>:serve-base:production"
      }
    }
  }
}
