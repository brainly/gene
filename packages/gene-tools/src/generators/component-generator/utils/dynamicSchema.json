{
  "$schema": "http://json-schema.org/schema",
  "id": "BrainlyCLIComponent",
  "title": "Brainly CLI component",
  "type": "object",
  "description": "Creates Brainly component",
  "properties": {
    "name": {
      "type": "string",
      "x-custom-default": "my-component",
      "minLength": 3,
      "description": "Name of the component",
      "x-custom-prompt": "What is the name of your component?"
    },
    "props": {
      "type": "string",
      "x-custom-sample": "id:string,content:string,items:array",
      "description": "Props of your component (separated by comma)",
      "x-custom-prompt": "What are props of your component? Separate them using comma, leave empty input for no props and do not count copy prop here."
    },
    "events": {
      "type": "string",
      "x-custom-sample": "on-click",
      "description": "Events names of your component (separated by comma)",
      "x-custom-prompt": "What are the events names of your component? Separate them using comma and leave empty input for no events."
    },
    "copy": {
      "type": "string",
      "x-custom-sample": "title,description",
      "description": "Copy keys of your component (separated by comma)",
      "x-custom-prompt": "What are the copy keys of your component? Separate them using comma and leave empty input for no copy."
    },
    "styles": {
      "type": "string",
      "x-custom-sample": "wrapper,element",
      "description": "Styles classes of your component (separated by comma)",
      "x-custom-prompt": "What are the style classes of your component? Separate them using comma and leave empty input for no classes."
    },
    "tests": {
      "type": "boolean",
      "x-custom-default": true,
      "x-custom-sample": true,
      "description": "Confirmation on creating tests file",
      "x-custom-prompt": "Should your component includes tests file?"
    },
    "storybook": {
      "type": "boolean",
      "x-custom-default": true,
      "x-custom-sample": true,
      "description": "Confirmation on creating storybook file",
      "x-custom-prompt": "Should your component includes storybook file?"
    },
    "reexport": {
      "type": "boolean",
      "x-custom-default": true,
      "x-custom-sample": true,
      "description": "Confirmation on reexporting",
      "x-custom-prompt": "Should your component be exported by a package?"
    },
    "data-testid": {
      "type": "string",
      "description": "[data-testid] parameter for QA and RTL testing",
      "x-custom-prompt": "What is the name of data-testid of your component? For no one, component name will be used."
    }
  },
  "required": []
}
