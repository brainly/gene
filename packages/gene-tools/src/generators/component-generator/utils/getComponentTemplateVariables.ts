import { pipe, map } from 'ramda';

import {
  splitStringToArray,
  splitStringToEntries,
  parsePropsEntries,
  defaultPropValue,
  defaultStorybookInput,
  getStorybookKnobs,
  isPrimitiveType,
} from './typesUtils';
import {
  classify,
  underscore,
  camelize,
  capitalize,
} from '@nx/devkit/src/utils/string-utils';

export const getComponentTemplateVariables = (
  _options: Record<string, any>,
) => {
  const {
    props,
    events,
    copy,
    styles,
    'data-testid': dataTestId,
    'facade-events-mediation': facadeEventsMediation,
    'sg-components': sgComponents,
    'markup-content': markupContent,
  } = _options;

  const eventsPrefix = 'Gene-';
  const flowDeclaration = false;
  const readonlyUtilType = 'Readonly';
  const shouldAddFlowDeclaration = false;
  const defaultPropsType = 'unknown';
  const stylesExtension = 'scss';

  const propsArray = pipe(
    splitStringToEntries(defaultPropsType) as any,
    map(
      parsePropsEntries({
        defaultValue: defaultPropsType,
        shouldAddFlowDeclaration,
      }),
    ),
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
  )(props as string);
  const eventsArray = splitStringToArray(events as string);
  const copyArray = splitStringToArray(copy as string);
  const stylesArray = splitStringToArray(styles as string);

  const storybookKnobs = getStorybookKnobs(propsArray, copyArray);
  const sgComponentsArray = splitStringToArray(sgComponents as string);

  const eventsObjectName = (name: string) =>
    `${underscore(name).toUpperCase()}_EVENTS_TYPE`;
  const eventsTypeName = (name: string) => `${classify(name)}EventsType`;
  const eventTypeName = (componentName: string, eventName: string) =>
    `${classify(componentName)}${classify(eventName)}Type`;
  const eventsKeyName = (event: string) => underscore(event).toUpperCase();
  const eventValueName = (event: string, name: string, eventsPrefix: string) =>
    `${eventsPrefix}${classify(name)}${classify(event)}`;
  const copyTypeName = (name: string) => `${classify(name)}CopyType`;
  const dataTestIdName = (name: string) => dataTestId || underscore(name);
  const valuesOfType = (typeName: string) =>
    `typeof ${typeName}[keyof typeof ${typeName}]`;

  return {
    propsArray,
    eventsArray,
    copyArray,
    stylesArray,
    readonlyUtilType,
    eventsPrefix,
    flowDeclaration,
    shouldAddFlowDeclaration,
    stylesExtension,
    storybookKnobs,
    facadeEventsMediation,
    sgComponents,
    sgComponentsArray,
    markupContent,
    valuesOfType,
    eventsObjectName,
    eventsTypeName,
    eventTypeName,
    eventsKeyName,
    eventValueName,
    copyTypeName,
    dataTestIdName,
    defaultPropValue,
    defaultStorybookInput,
    isPrimitiveType,
    classify,
    underscore,
    camelize,
    capitalize,
  };
};
