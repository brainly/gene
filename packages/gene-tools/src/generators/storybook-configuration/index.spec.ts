import * as devkit from '@nx/devkit';
import type { Tree } from '@nx/devkit';
import { logger, readJson, readProjectConfiguration } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { applicationGenerator } from '@nx/next';
import { libraryGenerator } from '@nx/react';
import { storybookConfigurationGenerator } from './index';
import { isCI } from 'nx/src/utils/is-ci';

jest.mock('@nx/devkit', () => {
  return {
    __esModule: true,
    ...jest.requireActual('@nx/devkit'),
  };
});

jest.setTimeout(30000); // NX fetches @nx/playwright with package manager during tests (to be mocked)

(isCI() ? describe.skip : describe)('storybookConfiguration generator', () => {
  let tree: Tree;

  beforeEach(async () => {
    jest.spyOn(logger, 'warn').mockImplementation(() => jest.fn());
    jest.spyOn(logger, 'debug').mockImplementation(() => jest.fn());

    // Create a tree with an empty workspace
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

    await new Promise(process.nextTick);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('application', () => {
    const application = 'my-app1';

    beforeEach(async () => {
      // Generate the application to add configuration to
      await applicationGenerator(tree, {
        name: application,
        directory: `apps/${application}`,
        linter: 'eslint',
        skipFormat: false,
        style: 'css',
        unitTestRunner: 'jest',
      });
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    it('should throw when passing a non-existent project', async () => {
      await expect(async () =>
        storybookConfigurationGenerator(tree, { name: 'non-existent-project' })
      ).rejects.toThrow();
    });

    it('should do nothing when the .storybook folder already exists and contains files', async () => {
      jest.spyOn(devkit, 'formatFiles');
      tree.write(`apps/${application}/.storybook/main.js`, '');
      const initialTreeChanges = tree.listChanges();

      await storybookConfigurationGenerator(tree, { name: application });

      expect(tree.listChanges()).toStrictEqual(initialTreeChanges);
      expect(devkit.formatFiles).not.toHaveBeenCalled();
    });

    it('should generate configuration files when .storybook folder exists without any files inside', async () => {
      jest.spyOn(devkit, 'formatFiles');
      tree.write(`apps/${application}/.storybook/`, '');

      await storybookConfigurationGenerator(tree, { name: application });

      expect(devkit.formatFiles).toHaveBeenCalled();
      expect(
        tree.read(`apps/${application}/.storybook/main.js`, 'utf-8')
      ).toMatchSnapshot();
      expect(
        tree.read(`apps/${application}/.storybook/manager.js`, 'utf-8')
      ).toMatchSnapshot();
      expect(
        tree.read(`apps/${application}/.storybook/preview.js`, 'utf-8')
      ).toMatchSnapshot();
      expect(
        tree.read(`apps/${application}/tsconfig.storybook.json`, 'utf-8')
      ).toMatchSnapshot();
      expect(tree.exists(`apps/${application}/.storybook/tsconfig.json`)).toBe(
        false
      );
    });

    it('should generate the .storybook folder with the configuration files', async () => {
      await storybookConfigurationGenerator(tree, { name: application });

      expect(
        tree.read(`apps/${application}/.storybook/main.js`, 'utf-8')
      ).toMatchSnapshot();
      expect(
        tree.read(`apps/${application}/.storybook/manager.js`, 'utf-8')
      ).toMatchSnapshot();
      expect(
        tree.read(`apps/${application}/.storybook/preview.js`, 'utf-8')
      ).toMatchSnapshot();
      expect(
        tree.read(`apps/${application}/tsconfig.storybook.json`, 'utf-8')
      ).toMatchSnapshot();
      expect(tree.exists(`apps/${application}/.storybook/tsconfig.json`)).toBe(
        false
      );
    });

    it('should exclude the stories from the project root tsconfig', async () => {
      await storybookConfigurationGenerator(tree, { name: application });

      const tsconfig = readJson(tree, `apps/${application}/tsconfig.json`);
      expect(tsconfig.exclude).toEqual(
        expect.arrayContaining([
          '**/*.stories.ts',
          '**/*.stories.js',
          '**/*.stories.jsx',
          '**/*.stories.tsx',
        ])
      );
    });

    it('should add storybook targets to the project configuration', async () => {
      await storybookConfigurationGenerator(tree, { name: application });

      const { targets } = readProjectConfiguration(tree, application);
      expect(targets && targets.storybook).toMatchSnapshot();
      expect(targets && targets['build-storybook']).toMatchSnapshot();
    });

    it('should format files', async () => {
      jest.spyOn(devkit, 'formatFiles');

      await storybookConfigurationGenerator(tree, { name: application });

      expect(devkit.formatFiles).toHaveBeenCalled();
    });
  });

  describe('library', () => {
    const library = 'my-lib1';

    beforeEach(async () => {
      // Generate the library to add configuration to
      await libraryGenerator(tree, {
        name: library,
        directory: `libs/${library}`,
        linter: 'eslint',
        skipFormat: false,
        skipTsConfig: false,
        style: 'css',
        unitTestRunner: 'jest',
      });

      jest.clearAllMocks();
    });

    it('should throw when passing a non-existent project', async () => {
      await expect(async () =>
        storybookConfigurationGenerator(tree, { name: 'non-existent-project' })
      ).rejects.toThrow();
    });

    it('should do nothing when the .storybook folder already exists', async () => {
      jest.spyOn(devkit, 'formatFiles');
      tree.write(`libs/${library}/.storybook/main.js`, '');
      const initialTreeChanges = tree.listChanges();

      await storybookConfigurationGenerator(tree, { name: library });

      expect(tree.listChanges()).toStrictEqual(initialTreeChanges);
      expect(devkit.formatFiles).not.toHaveBeenCalled();
    });

    it('should generate the .storybook folder with the configuration files', async () => {
      await storybookConfigurationGenerator(tree, { name: library });

      expect(
        tree.read(`libs/${library}/.storybook/main.js`, 'utf-8')
      ).toMatchSnapshot();
      expect(
        tree.read(`libs/${library}/.storybook/manager.js`, 'utf-8')
      ).toMatchSnapshot();
      expect(
        tree.read(`libs/${library}/.storybook/preview.js`, 'utf-8')
      ).toMatchSnapshot();
      expect(
        tree.read(`libs/${library}/tsconfig.storybook.json`, 'utf-8')
      ).toMatchSnapshot();
      expect(tree.exists(`libs/${library}/.storybook/tsconfig.json`)).toBe(
        false
      );
    });

    it('should exclude the stories from the project root tsconfig', async () => {
      await storybookConfigurationGenerator(tree, { name: library });

      const tsconfig = readJson(tree, `libs/${library}/tsconfig.lib.json`);
      expect(tsconfig.exclude).toEqual(
        expect.arrayContaining([
          '**/*.stories.ts',
          '**/*.stories.js',
          '**/*.stories.jsx',
          '**/*.stories.tsx',
        ])
      );
    });

    it('should add storybook targets to the project configuration', async () => {
      await storybookConfigurationGenerator(tree, { name: library });

      const { targets } = readProjectConfiguration(tree, library);
      expect(targets && targets.storybook).toMatchSnapshot();
      expect(targets && targets['build-storybook']).toMatchSnapshot();
    });

    it('should format files', async () => {
      jest.spyOn(devkit, 'formatFiles');

      await storybookConfigurationGenerator(tree, { name: library });

      expect(devkit.formatFiles).toHaveBeenCalled();
    });
  });
});
