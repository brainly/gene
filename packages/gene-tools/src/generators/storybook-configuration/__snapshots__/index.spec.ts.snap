// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storybookConfiguration generator application should add storybook targets to the project configuration 1`] = `
Object {
  "executor": "@brainly/gene-tools:storybook",
  "options": Object {
    "command": "start",
  },
}
`;

exports[`storybookConfiguration generator application should add storybook targets to the project configuration 2`] = `
Object {
  "executor": "@brainly/gene-tools:storybook",
  "options": Object {
    "command": "build",
  },
}
`;

exports[`storybookConfiguration generator application should generate configuration files when .storybook folder exists without any files inside 1`] = `
"const rootMain = require('../../../.storybook/main');

module.exports = {
  ...rootMain,
  stories: [
    
    '../components/**/*.stories.mdx',
    '../components/**/*.stories.@(js|jsx|ts|tsx)',
    
  ],

  webpackFinal: async (config, {configType}) => {
    // apply any global webpack configs that might have been specified in .storybook/main.js
    if (rootMain.webpackFinal) {
      config = await rootMain.webpackFinal(config, {configType});
    }

    // add your own webpack tweaks if needed

    return config;
  },
};
"
`;

exports[`storybookConfiguration generator application should generate configuration files when .storybook folder exists without any files inside 2`] = `
"import '../../../.storybook/manager';
"
`;

exports[`storybookConfiguration generator application should generate configuration files when .storybook folder exists without any files inside 3`] = `
"import {workerPath} from '../../../.storybook/preview';
import {initialize} from 'msw-storybook-addon';
const {use} = initialize({
  serviceWorker: {
    url: workerPath,
  },
});

export {use};

export {decorators, parameters} from '../../../.storybook/preview';
"
`;

exports[`storybookConfiguration generator application should generate configuration files when .storybook folder exists without any files inside 4`] = `
"{
  \\"extends\\": \\"../tsconfig.json\\",
  \\"compilerOptions\\": {
    \\"emitDecoratorMetadata\\": true,
    \\"experimentalDecorators\\": true,
    \\"outDir\\": \\"\\"
  },
  \\"exclude\\": [
    \\"../**/*.spec.ts\\",
    \\"../**/*.spec.js\\",
    \\"../**/*.spec.tsx\\",
    \\"../**/*.spec.jsx\\"
  ],
  \\"include\\": [
    \\"../**/*.stories.ts\\",
    \\"../**/*.stories.js\\",
    \\"../**/*.stories.jsx\\",
    \\"../**/*.stories.tsx\\",
    \\"*.js\\"
  ]
}
"
`;

exports[`storybookConfiguration generator application should generate the .storybook folder with the configuration files 1`] = `
"const rootMain = require('../../../.storybook/main');

module.exports = {
  ...rootMain,
  stories: [
    
    '../components/**/*.stories.mdx',
    '../components/**/*.stories.@(js|jsx|ts|tsx)',
    
  ],

  webpackFinal: async (config, {configType}) => {
    // apply any global webpack configs that might have been specified in .storybook/main.js
    if (rootMain.webpackFinal) {
      config = await rootMain.webpackFinal(config, {configType});
    }

    // add your own webpack tweaks if needed

    return config;
  },
};
"
`;

exports[`storybookConfiguration generator application should generate the .storybook folder with the configuration files 2`] = `
"import '../../../.storybook/manager';
"
`;

exports[`storybookConfiguration generator application should generate the .storybook folder with the configuration files 3`] = `
"import {workerPath} from '../../../.storybook/preview';
import {initialize} from 'msw-storybook-addon';
const {use} = initialize({
  serviceWorker: {
    url: workerPath,
  },
});

export {use};

export {decorators, parameters} from '../../../.storybook/preview';
"
`;

exports[`storybookConfiguration generator application should generate the .storybook folder with the configuration files 4`] = `
"{
  \\"extends\\": \\"../tsconfig.json\\",
  \\"compilerOptions\\": {
    \\"emitDecoratorMetadata\\": true,
    \\"experimentalDecorators\\": true,
    \\"outDir\\": \\"\\"
  },
  \\"exclude\\": [
    \\"../**/*.spec.ts\\",
    \\"../**/*.spec.js\\",
    \\"../**/*.spec.tsx\\",
    \\"../**/*.spec.jsx\\"
  ],
  \\"include\\": [
    \\"../**/*.stories.ts\\",
    \\"../**/*.stories.js\\",
    \\"../**/*.stories.jsx\\",
    \\"../**/*.stories.tsx\\",
    \\"*.js\\"
  ]
}
"
`;

exports[`storybookConfiguration generator library should add storybook targets to the project configuration 1`] = `
Object {
  "executor": "@brainly/gene-tools:storybook",
  "options": Object {
    "command": "start",
  },
}
`;

exports[`storybookConfiguration generator library should add storybook targets to the project configuration 2`] = `
Object {
  "executor": "@brainly/gene-tools:storybook",
  "options": Object {
    "command": "build",
  },
}
`;

exports[`storybookConfiguration generator library should generate the .storybook folder with the configuration files 1`] = `
"const rootMain = require('../../../.storybook/main');

module.exports = {
  ...rootMain,
  stories: [
    
    '../src/**/*.stories.mdx',
    '../src/**/*.stories.@(js|jsx|ts|tsx)',
    
  ],

  webpackFinal: async (config, {configType}) => {
    // apply any global webpack configs that might have been specified in .storybook/main.js
    if (rootMain.webpackFinal) {
      config = await rootMain.webpackFinal(config, {configType});
    }

    // add your own webpack tweaks if needed

    return config;
  },
};
"
`;

exports[`storybookConfiguration generator library should generate the .storybook folder with the configuration files 2`] = `
"import '../../../.storybook/manager';
"
`;

exports[`storybookConfiguration generator library should generate the .storybook folder with the configuration files 3`] = `
"import {workerPath} from '../../../.storybook/preview';
import {initialize} from 'msw-storybook-addon';
const {use} = initialize({
  serviceWorker: {
    url: workerPath,
  },
});

export {use};

export {decorators, parameters} from '../../../.storybook/preview';
"
`;

exports[`storybookConfiguration generator library should generate the .storybook folder with the configuration files 4`] = `
"{
  \\"extends\\": \\"../tsconfig.json\\",
  \\"compilerOptions\\": {
    \\"emitDecoratorMetadata\\": true,
    \\"experimentalDecorators\\": true,
    \\"outDir\\": \\"\\"
  },
  \\"exclude\\": [
    \\"../**/*.spec.ts\\",
    \\"../**/*.spec.js\\",
    \\"../**/*.spec.tsx\\",
    \\"../**/*.spec.jsx\\"
  ],
  \\"include\\": [
    \\"../**/*.stories.ts\\",
    \\"../**/*.stories.js\\",
    \\"../**/*.stories.jsx\\",
    \\"../**/*.stories.tsx\\",
    \\"*.js\\"
  ]
}
"
`;
